File: console/src/main/kotlin/app/accrescent/parcelo/console/data/net/Update.kt
// Copyright 2023-2024 Logan Magee
//
// SPDX-License-Identifier: AGPL-3.0-only

package app.accrescent.parcelo.console.data.net

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Update(
    val id: String,
    @SerialName("app_id")
    val appId: String,
    @SerialName("version_code")
    val versionCode: Int,
    @SerialName("version_name")
    val versionName: String,
    @SerialName("creation_time")
    val creationTime: Long,
    @SerialName("requires_review")
    val requiresReview: Boolean,
    val status: UpdateStatus,
)

@Serializable
enum class UpdateStatus {
    @SerialName("unsubmitted")
    UNSUBMITTED,

    @SerialName("pending-review")
    PENDING_REVIEW,

    @SerialName("rejected")
    REJECTED,

    @SerialName("publishing")
    PUBLISHING,

    @SerialName("published")
    PUBLISHED,
}

File: console/frontend/src/app/update.ts
// Copyright 2023 Logan Magee
//
// SPDX-License-Identifier: AGPL-3.0-only

export interface Update {
    id: string;
    app_id: string;
    version_code: number;
    version_name: string;
    creation_time: number;
    requires_review: boolean;
    status: UpdateStatus;
}

export enum UpdateStatus {
    Unsubmitted = 'unsubmitted',
    PendingReview = 'pending-review',
    Publishing = 'publishing',
    Published = 'published',
}
